igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/client/model/Plan.java

```diff
@@ -16,73 +16,56 @@
 
 package com.netflix.spinnaker.igor.concourse.client.model;
 
-import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.stream.Collectors;
-import javax.annotation.Nullable;
+import java.util.Map.Entry;
 import lombok.Data;
-import lombok.Setter;
 
 @Data
 public class Plan {
-  private InnerPlan plan;
+  private ObjectNode plan;
 
-  @Data
-  public static class InnerPlan {
-    @JsonAlias("do")
-    private List<Op> does;
-
-    @Nullable private OnFailure onFailure;
+  public List<Resource> getResources() {
+    return getResources(plan);
   }
 
-  @Setter
-  public static class Op {
-    private String id;
+  private List<Resource> getResources(ObjectNode o) {
+    if (o.hasNonNull("id")) {
+      if (o.hasNonNull("get")) {
+        return Collections.singletonList(parseResource(o.get("id").asText(), o.get("get")));
+      }
+      if (o.hasNonNull("put")) {
+        return Collections.singletonList(parseResource(o.get("id").asText(), o.get("put")));
+      }
+    }
 
-    @Nullable private ResourceShape get;
+    List<Resource> res = new ArrayList<>();
 
-    @Nullable private ResourceShape put;
+    Iterator<Entry<String, JsonNode>> fields = o.fields();
+    while (fields.hasNext()) {
+      Entry<String, JsonNode> f = fields.next();
 
-    @Nullable private OnSuccess onSuccess;
+      if (f.getValue().isArray()) {
+        for (int i = 0; i < f.getValue().size(); i++) {
+          JsonNode jsonNode = f.getValue().get(i);
+          if (jsonNode.isObject()) {
+            res.addAll(getResources((ObjectNode) jsonNode));
+          }
```

Comment:
Nitpick, put could this be replaced with a stream instead?
```suggestion
        StreamSupport.stream(f.getValue().spliterator(), false)
            .filter(JsonNode::isObject)
            .map(jsonNode -> getResources((ObjectNode) jsonNode))
            .forEach(res::addAll);
```

Reply:
done